nixos/modules/services/mail/dovecot.nix

```diff
@@ -467,10 +467,6 @@ in
     ];
 
     assertions = [
-      {
```

Comment:
Maybe we could reword this and turn it into a warning instead? It does seem reasonable for most use-cases, doesn't it?

Side-note: Digging through the history, this assertion has been in there forever™ (before 5c1f8cbc70cd5e6867ef6a2a06d27a40daa07010)

Reply:
To be honest, the module has imap already enabled by default. So to trip this assertion you have to explicitly disable imap and at that point it's more than likely you don't want to fetch mail from that server, unless you meant to only allow POP3 but didn't know to enable it - perhaps because you assumed that `pop3` was enabled by default too.

Making this a warning creates would create noise every time we build this module for our smarthosts, our relays and our submission servers, I'd not be happy with this.

To me it seems wanting to run an authenticating mail relay/submission server is way more likely than wanting to run a dedicated POP3 server...

Reply:
OTOH originally I wanted something like this in our common server role:
```nix
{
  config = {
    services.dovecot2.enableImap = lib.mkForce false;
    services.dovecot2.enablePop3 = lib.mkForce false;

    # I consider this to be a poor default!
    services.dovecot2.enablePAM = lib.mkForce false;
  };
}
```

And then consistently just use `services.dovecot2.protocols`. It feels more natural to me.

Reply:
Fair enough. What do you run dovecot on your submission and smarthost hosts for in the first place? Just auth?

The enablePAM thing might also be worth flipping the default on, but this is getting off-topic.

Back on-topic: The dovecot module has had various contributors over the years, so let's give some of the more recent ones e.g. @symphorien and @Ma27 a few days to also take a look at this. If neither of them decide to comment on this, we can probably just go ahead with it as is, since it "just" removes an assertion.

Reply:
Yes, we run local dovecot on those servers just for SASL authentication.